source/apiVolontaria/volunteer/tests/tests_view_EventsId.py

```diff
@@ -0,0 +1,267 @@
+            strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+        end_date_str = self.event.end_date.\
+            strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+
+        data = {
+            'id': self.event.id,
+            'cell': self.event.cell.id,
+            'cycle': self.event.cycle.id,
+            'task_type': self.event.task_type.id,
+            'start_date': start_date_str,
+            'end_date': end_date_str,
+            'nb_volunteers_needed': 10,
+            'nb_volunteers_standby_needed':
+                self.event.nb_volunteers_standby_needed,
+            'volunteers': [],
+            'volunteers_standby': [],
+        }
+
+        data_post = {
+            "nb_volunteers_needed": 10,
+
+        }
+
+        self.client.force_authenticate(user=self.admin)
+
+        response = self.client.patch(
+            reverse(
+                'volunteer:events_id',
+                kwargs={'pk': self.event.id},
+            ),
+            data_post,
+            format='json',
+        )
+
+        self.assertEqual(json.loads(response.content), data)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_update_event_without_permission(self):
+        """
+        Ensure we can't update a specific event without permission.
+        """
+        data_post = {
+            "nb_volunteers_needed": 10,
+
+        }
+
+        self.client.force_authenticate(user=self.user)
+
+        response = self.client.patch(
+            reverse(
+                'volunteer:events_id',
+                kwargs={'pk': self.event.id},
+            ),
+            data_post,
+            format='json',
+        )
+
+        content = {'detail': "You are not authorized to update an event."}
+        self.assertEqual(json.loads(response.content), content)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_update_event_that_doesnt_exist(self):
+        """
+        Ensure we can't update a specific event if it doesn't exist.
+        """
+        data_post = {
+            "nb_volunteers_needed": 10,
+        }
+
+        self.client.force_authenticate(user=self.admin)
+
+        response = self.client.patch(
+            reverse(
+                'volunteer:events_id',
+                kwargs={'pk': 9999},
+            ),
+            data_post,
+            format='json',
+        )
+
+        content = {'detail': "Not found."}
+
+        self.assertEqual(json.loads(response.content), content)
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_delete_event_with_permission(self):
+        """
+        Ensure we can delete a specific event.
+        """
+        self.client.force_authenticate(user=self.admin)
+
+        response = self.client.delete(
+            reverse(
+                'volunteer:events_id',
+                kwargs={'pk': self.event.id},
+            ),
+        )
+
+        self.assertEqual(response.content, b'')
```

Comment:
Why not `json.loads()` ?

Reply:
Because it's not a JSON, it's just empty because we have a `HTTP_204_NO_CONTENT`